sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -230,6 +230,12 @@ public virtual Response<TableItem> Create(CancellationToken cancellationToken =
                 var response = _tableOperations.Create(new TableProperties() { TableName = _table }, null, queryOptions: _defaultQueryOptions, cancellationToken: cancellationToken);
                 return Response.FromValue(response.Value as TableItem, response.GetRawResponse());
             }
+            catch (RequestFailedException rfe)
+            {
+                var ex = TablesExceptionParser.Parse(rfe);
+                scope.Failed(ex);
+                throw ex;
```

Comment:
Should we consider using `ExceptionDispatchInfo` here to throw, in case the `rfe` isn't translated by the parser?   Looks like that can happen when it's not an OData error...